tests/test_schema.py

```diff
@@ -30,13 +30,57 @@ def load_json(filename):
     return doc
 
 
+def test_delivery():
+    """Test delivery superset schema."""
+    upsert = load_json('valid.json')
+    takedown = copy.deepcopy(upsert)
+    takedown['action'] = 'takedown'
+    assert schema.validate_schema(schema.delivery, upsert) == upsert
+    assert schema.validate_schema(schema.delivery, takedown) == takedown
+
+
 def test_empty_document():
     """Test that an empty document doesn't validate."""
     doc = load_json('empty.json')
     with pytest.raises(schema.MultipleInvalid):
         schema.track_bundle(doc)
 
 
+@pytest.mark.parametrize('type_', (
+    'AdvertsiementSupportedModel',
```

Comment:
I'm not a fan of typos like this as part of test data. It can be really hard to spot the difference between a valid value and an invalid one. The same applies to `PayAsUGoModel`.